Linking fails on dev-2024-05 with: cannot find crtbegin.o

## Problem
`./build_odin.sh` succeeds on dev-2024-04a and prints the demo.

`./build_odin.sh` fails on dev-2024-05 with the following output:

```bash
~/Documents/odin $ ./build_odin.sh 
+ /usr/lib/llvm17/bin/clang++ src/main.cpp src/libtommath.cpp -Wno-switch -Wno-macro-redefined -Wno-unused-value '-DODIN_VERSION_RAW="dev-2024-05"' '-DGIT_SHA="2250eb3e7"' '-std=c++14' -I/usr/lib/llvm17/include '-std=c++17' -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -L/usr/lib/llvm17/lib -g -pthread -lm -lstdc++ -ldl /usr/lib/llvm17/lib/libLLVM-17.so '-Wl,-rpath=$ORIGIN' -o odin
+ set +x
/usr/bin/ld: cannot find crtbegin.o: No such file or directory
/usr/bin/ld: cannot find -lgcc: No such file or directory
/usr/bin/ld: cannot find -lgcc: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Somewhere between dev-2024-04a and dev-2024-05 an issue was introduced, or have the build requirements changed?

## System
OS: Alpine Linux edge, Linux 6.6.30-0-lts
Arch: x86_64
Libc: musl

## Version
Odin: dev-2024-05:2250eb3e7
Backend: LLVM 17.0.6

--- a/src/build_settings.cpp
+++ b/src/build_settings.cpp
@@ -2035,6 +2035,9 @@ gb_internal void init_build_context(TargetMetrics *cross_target, Subtarget subta
 			bc->link_flags = str_lit("/machine:x86 ");
 			break;
 		}
+	} else if (bc->metrics.os == TargetOs_darwin) {
+		bc->link_flags = concatenate3_strings(permanent_allocator(),
+			str_lit("-target "), bc->metrics.target_triplet, str_lit(" "));
 	} else if (is_arch_wasm()) {
 		gbString link_flags = gb_string_make(heap_allocator(), " ");
 		// link_flags = gb_string_appendc(link_flags, "--export-all ");
@@ -2052,8 +2055,13 @@ gb_internal void init_build_context(TargetMetrics *cross_target, Subtarget subta
 		// Disallow on wasm
 		bc->use_separate_modules = false;
 	} else {
-		bc->link_flags = concatenate3_strings(permanent_allocator(),
-			str_lit("-target "), bc->metrics.target_triplet, str_lit(" "));
+		// NOTE: for targets other than darwin, we don't specify a `-target` link flag.
+		// This is because we don't support cross-linking and clang is better at figuring
+		// out what the actual target for linking is,
+		// for example, on x86/alpine/musl it HAS to be `x86_64-alpine-linux-musl` to link correctly.
+		//
+		// Note that codegen will still target the triplet we specify, but the intricate details of
+		// a target shouldn't matter as much to codegen (if it does at all) as it does to linking.
 	}
 
 	// NOTE: needs to be done after adding the -target flag to the linker flags so the linker

